src/libstd/sys/hermit/stdio.rs

```diff
@@ -10,19 +10,21 @@ impl Stdin {
     pub fn new() -> io::Result<Stdin> {
         Ok(Stdin)
     }
+}
 
-    pub fn read(&self, data: &mut [u8]) -> io::Result<usize> {
+impl io::Read for Stdin {
+    fn read(&mut self, data: &mut [u8]) -> io::Result<usize> {
         self.read_vectored(&mut [IoSliceMut::new(data)])
     }
 
-    pub fn read_vectored(&self, _data: &mut [IoSliceMut<'_>]) -> io::Result<usize> {
+    fn read_vectored(&self, _data: &mut [IoSliceMut<'_>]) -> io::Result<usize> {
         //ManuallyDrop::new(unsafe { WasiFd::from_raw(libc::STDIN_FILENO as u32) })
         //    .read(data)
```

Comment:
What is this comment about? According to git blame it was introduced like this. https://github.com/rust-lang/rust/commit/c1e440a90f472468c8069ba6254b23c6feedc32e#diff-4e8bc140a84b306d4956ad6e2f5c81fd

Can you elaborate why this code is there @stlankes 
Can it be removed?

Reply:
Yes, I will remove it. Sorry, I oversaw it.

Reply:
Do you want me to do that in this PR? It's not really in scope, but there's no reason to risk a future merge conflict

Reply:
Got it

Reply:
Yes, this will be great. Just remove the line.